### PR TITLE
[DX][FrameworkBundle][Server Command] Launch browser window when server is started

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | ~ |
| Tests pass? | yes |
| License | MIT |

I tested only on `ubuntu`

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php [hunk 1]
```diff
@@ -154,6 +155,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
             unlink($lockFile);
 
             return 1;
+        } elseif ($input->getOption('launch-browser')) {
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php]: no need for `else` here

Comment (aitboudad) [inline src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php]: fixed

File: src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php [hunk 2]
```diff
@@ -244,4 +249,20 @@ private function createServerProcess(OutputInterface $output, $address, $documen
 
         return new Process('exec '.$script, $documentRoot, null, null, null);
     }
+
+    private function launchBrowser($output, $address)
+    {
+        $cmdOpen = 'xdg-open';
+        if (stristr(PHP_OS, 'WIN')) {
```
Comment (pborreli) [inline src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php]: for consistency with the rest of the Symfony codebase I would use `stripos` here

Comment (aitboudad) [inline src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php]: done

File: src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php [hunk 3]
```diff
@@ -244,4 +249,20 @@ private function createServerProcess(OutputInterface $output, $address, $documen
 
         return new Process('exec '.$script, $documentRoot, null, null, null);
     }
+
+    private function launchBrowser($output, $address)
+    {
+        $cmdOpen = 'xdg-open';
+        if (false !== stripos(PHP_OS, 'win')) {
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php]: To detect Windows builds I would just check the value of the `DIRECTORY_SEPARATOR` constant as we do everywhere else:

```php
'\\' === DIRECTORY_SEPARATOR
```

File: src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php [hunk 4]
```diff
@@ -244,4 +249,20 @@ private function createServerProcess(OutputInterface $output, $address, $documen
 
         return new Process('exec '.$script, $documentRoot, null, null, null);
     }
+
+    private function launchBrowser($output, $address)
+    {
+        $cmdOpen = 'xdg-open';
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Command/ServerStartCommand.php]: This could be moved to an `else`.

**Comment by fabpot** [comment]
Not sure this is really useful. People now needs to add this long option to open the browser vs just clicking on the URL from the console (your console supports this right?). So, I'm :-1:

**Comment by aitboudad** [comment]
@fabpot 
>People now needs to add this long option to open the browser

use the shortcut value solves the issue `app/console server:start -b` 

>clicking on the URL from the console

not supported in all OS

**Comment by aitboudad** [comment]
This is meant to be a DX Initiative improvement. 
@javiereguiluz any thoughts? or should be closed ?

**Comment by fabpot** [comment]
Closing this as I find it not that useful, adds yet another option, and most of the time, your browser is already open anyway.

**Pull request closed without merging**